apps/admin/src/query/peak.ts

```diff
@@ -0,0 +1,64 @@
+"use server";
+
+import { getPeaksWithoutDevices, logError, saveDeviceToPeakDb } from "@energyleaf/db/query";
+import "server-only";
+import { mlApi } from "@/actions/ml";
+import { waitUntil } from "@vercel/functions";
+
+const deviceNameMapping = {
+    fridge: "Kühlschrank",
+    freezer: "Gefrierschrank",
+    micro_wave_oven: "Mikrowelle",
+    dishwasher: "Spülmaschine",
+    washing_machine: "Waschmaschine",
+};
+
+export async function classifyAndSaveDevicesForPeaks(
+    peaks: { id: string; electricity: { timestamp: string; power: number }[] }[],
+    userId: string,
+) {
+    try {
+        const peaksToClassifyIds = await getPeaksWithoutDevices(peaks.map((peak) => ({ id: peak.id })));
+
+        if (peaksToClassifyIds.length === 0) {
+            return;
+        }
+
+        const peaksToClassify = peaks.filter((peak) =>
+            peaksToClassifyIds.some((peakToClassify) => peakToClassify.id === peak.id),
+        );
+
+        const response = await mlApi({ peaks: peaksToClassify });
+
+        for (const peak of response.peaks) {
+            const devicesToSave = peak.devices
+                .filter((device) => device.confidence >= 0.9)
+                .map((device) => ({
+                    name: deviceNameMapping[device.name] || device.name,
+                    confidence: device.confidence,
+                }));
+
+            if (devicesToSave.length > 0) {
+                await saveDevicesToPeak(peak.id, devicesToSave, userId);
```

Comment:
Wenn ich den Code richtig verstehe, sind die automatisiert zugewiesenen Geräte doch gar nicht in der DB gespeichert, oder?